src/helpers.rs

```diff
@@ -1017,6 +1017,35 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         Ok((true, string_length))
     }
 
+    /// Read a sequence of wchar_t until the first null terminator.
+    /// Always returns a `Vec<u32>` no matter the size of `wchar_t`.
+    fn read_wchar_t_str(
+        &self,
+        mut ptr: Pointer<Option<Provenance>>,
+    ) -> InterpResult<'tcx, Vec<u32>> {
+        let this = self.eval_context_ref();
+
+        let wchar_t = this.libc_ty_layout("wchar_t");
+
+        this.check_ptr_align(ptr, wchar_t.align.abi)?;
+
+        let mut wchars = Vec::new();
+        loop {
+            // FIXME: We are re-getting the allocation each time around the loop.
+            // Would be nice if we could somehow "extend" an existing AllocRange.
+            let alloc = this.get_ptr_alloc(ptr, wchar_t.size)?.unwrap(); // not a ZST, so we will get a result
+            let wchar = alloc.read_integer(alloc_range(Size::ZERO, wchar_t.size))?.to_u32()?;
```

Comment:
the `to_u32` expects 4 bytes, but on windows it only has 2

> type for wide character representation (see [wide strings](https://en.cppreference.com/w/cpp/string/wide)). It has the same size, signedness, and alignment as one of the integer types, but is a distinct type. In practice, it is 32 bits and holds UTF-32 on Linux and many other non-Windows systems, but 16 bits and holds UTF-16 code units on Windows. The standard used to require wchar_t to be large enough to represent any supported character code point. However, such requirement cannot be fulfilled on Windows, and thus it is considered as a [defect](https://en.cppreference.com/w/cpp/language/types#Defect_reports) and removed.

you'll need to use `to_bits`

Reply:
You can grep for `off_t` to find how existing code handles "read an integer type that has platform-defined size".